### DISCUSSION TITLE
Mix just syntax with bash in a rule

### DISCUSSION THREAD

**Discussion by mihaigalos** [casey/just#880] [Q&A] [answered]
Hi there,

What is standard practice for mixing in a just variable into bash in the following code?:
```python
build tool:
    docker_image := docker_container_registry + "/" + {{tool}}
    #! /bin/bash
    docker build -t {{docker_image}} .
```

Here, I would like to compose `docker_image` independently of the bash script but then use it in the script.

**Comment by casey** [comment, marked as answer]
It currently isn't possible to create `just` variables inside a recipe. However, since you're using a shebang, you could do this:

```
build tool:
    #!/usr/bin/env bash
    docker_image={{ docker_container_registry }}/{{ tool }}
    docker build -t $docker_image .
```

**Reply by mihaigalos** [reply]
Thank you! :)

**Reply by casey** [reply]
You are most welcome!

**Reply by casey** [reply]
I edited my answer to be a little nicer. Also, I just released v0.9.6, which has a new `join` function, so you can do:

```
build tool:
    #!/usr/bin/env bash
    docker_image={{ join(docker_container_registry, tool) }}
    docker build -t $docker_image .
```

**Reply by mihaigalos** [reply]
Very nice. Awesome tool. Kudos!

**Reply by mihaigalos** [reply]
btw, I guess a shebang is not necessary anymore if `$docker_image` is not used afterwards:
```just
build tool:
    docker build -t {{ join(docker_container_registry, tool) }} .
```
